### DISCUSSION TITLE
slidesToScroll: 'auto' and slider ending with last slide at right edge instead of left edge

### DISCUSSION THREAD

**Discussion by fabz1001** [davidjerleke/embla-carousel#397] [Help] [answered]
Hi David and all the Embla supporters!

First of all, I appreciate your amazing work. Embla is great and I think I've finally found my favorite slider :)

So, I have a question: 
I needed to use the `slidesToScroll: 'auto'` option for a slider of buttons. I noticed the behaviour is a bit different than I expected (unless I'm doing something wrong!). I was expecting the slider to end with the last slide at the right edge of the viewport, although it seems to be ending at the beginning of it. 
I'm not sure I'm explaining it correctly with words, therefore I am attaching a screenshot as well as a [codesandbox](https://codesandbox.io/s/embla-carousel-default-react-forked-cn9lvi). 
In case this is intended and I'm not doing anything wrong, do you have any plans to adjust it to be that way?

Thank you very much in advance and feel free to respond whenever you have time :)

![embla](https://user-images.githubusercontent.com/57601672/202008333-383b2602-2e09-40e8-868a-874bd0b38e97.png)

**Comment by davidjerleke** [comment, marked as answer]
Hi @fabz1001,

Thank you for your question. I think you’re looking for the `containScroll: 'trimSnaps'` option. Add it to the carousel options and read more about it [here](https://www.embla-carousel.com/api/options/#containscroll):

```js
const [viewportRef, embla] = useEmblaCarousel({
  align: "start",
  slidesToScroll: "auto",
  loop: false,
  containScroll: "trimSnaps", // <— Add this
});
```

Please don’t forget to mark this comment as the answer if it solves your problem.

Best,
David

**Reply by fabz1001** [reply]
No way, how could I have missed that?
Thanks so much, David! :) Also for the blazing fast response, much appreciated. 

All the best,
Fabio

P.S.: please, don't take the donation as a compensation for the answer, I was planning to do it anyway.

**Reply by davidjerleke** [reply]
Thanks a lot @fabz1001!

Unfortunately, the documentation website isn’t the best. I’m working on a tool that will help devs setup and customize their carousels with ease. And hopefully, it will handle these kind of things for you. As soon as I have something working, I will add it to the documentation website.

Your support makes it possible for me to spend some quality time on this.

Thank you!

Best,
David
